File: src/main/java/com/project/pantau/dto/auth/UserResponse.java
package com.project.pantau.dto.auth;

import com.project.pantau.enums.UserRole;
import lombok.Builder;

import java.time.LocalDateTime;
import java.util.UUID;

@Builder
public record UserResponse(
        UUID uuid,
        String email,
        String displayName,
        UserRole role,
        LocalDateTime createdAt,
        LocalDateTime updatedAt
) {
}

File: backend/internal/dto/auth/user_response.go
package auth

import (
	"github.com/google/uuid"
	"pantau/internal/enums"
	"time"
)

type UserResponse struct {
	ID          uuid.UUID      `json:"id"`
	Email       string         `json:"email"`
	DisplayName string         `json:"display_name"`
	Role        enums.UserRole `json:"role"`
	CreatedAt   time.Time      `json:"created_at"`
	UpdatedAt   time.Time      `json:"updated_at"`
}
